src/components/Liveness/index.js

```diff
@@ -3,25 +3,32 @@ import * as React from 'react'
 import { h, Component } from 'preact'
 import Error from '../Error'
 import Spinner from '../Spinner'
-import { getLivenessChallenges } from '../utils/onfidoApi'
 import LivenessCamera from '../Camera/LivenessCamera'
 import type { CameraType } from '../Camera/CameraTypes'
 import type { ChallengeType } from './Challenge'
+import { performHttpReq } from '../utils/http'
 
 const serverError = { name: 'SERVER_ERROR', type: 'error' }
 
 type State = {
-  challenges: ChallengeType[],
+  id: string,
+  challenges: Array<ChallengeType>,
   hasError: boolean,
   hasLoaded: boolean,
+  startedAt: number,
+  switchSeconds?: number,
 };
 
 const initialState = {
+  id: '',
   challenges: [],
   hasLoaded: false,
   hasError: false,
+  startedAt: 0
 };
 
+const now = () => Date.now() / 1000 | 0
```

Comment:
You're just trying to do a `Math.floor` basically, right? I'd just go for that - it's clearer

Reply:
I tried using the same approach used here:
https://github.com/onfido/onfido-sdk-ui/blob/master/src/components/utils/jwt.js#L9

But yours is a fair point too. Do you think it's worth moving this to utils?

Reply:
I think so. `Math.floor` is hugely more understandable than bitwise operations